contracts/hooks/TotalLicenseTokenLimitHook.sol

```diff
@@ -0,0 +1,182 @@
+import { AccessControlled } from "@storyprotocol/core/access/AccessControlled.sol";
+import { ILicensingHook } from "@storyprotocol/core/interfaces/modules/licensing/ILicensingHook.sol";
+import { ILicenseRegistry } from "@storyprotocol/core/interfaces/registries/ILicenseRegistry.sol";
+import { ILicenseToken } from "@storyprotocol/core/interfaces/ILicenseToken.sol";
+import { ILicenseTemplate } from "@storyprotocol/core/interfaces/modules/licensing/ILicenseTemplate.sol";
+
+contract TotalLicenseTokenLimitHook is BaseModule, AccessControlled, ILicensingHook {
+    string public constant override name = "TotalLicenseTokenLimitHook";
+
+    ILicenseRegistry public immutable LICENSE_REGISTRY;
+    ILicenseToken public immutable LICENSE_TOKEN;
+    // keccak256(licensorIpId, licenseTemplate, licenseTermsId) => totalLicenseTokenLimit
+    mapping(bytes32 => uint256) public totalLicenseTokenLimit;
+
+    /// @notice Emitted when the total license token limit is set
+    /// @param licensorIpId The licensor IP id
+    /// @param licenseTemplate The license template address
+    /// @param licenseTermsId The license terms id
+    /// @param limit The total license token limit for the specific license of the licensor IP
+    event SetTotalLicenseTokenLimit(
+        address indexed licensorIpId,
+        address indexed licenseTemplate,
+        uint256 indexed licenseTermsId,
+        uint256 limit
+    );
+
+    /// @notice Emitted when the total license token limit is exceeded
+    /// @param totalSupply The total supply of the license tokens
+    /// @param amount The amount of license tokens to mint
+    /// @param limit The total license token limit
+    error TotalLicenseTokenLimitExceeded(uint256 totalSupply, uint256 amount, uint256 limit);
+
+    constructor(
+        address licenseRegistry,
+        address licenseToken,
+        address accessController,
+        address ipAssetRegistry
+    ) AccessControlled(accessController, ipAssetRegistry) {
+        require(licenseRegistry != address(0), "TotalLicenseTokenLimitHook: licenseRegistry is the zero address");
+        require(licenseToken != address(0), "TotalLicenseTokenLimitHook: licenseToken is the zero address");
+        LICENSE_REGISTRY = ILicenseRegistry(licenseRegistry);
+        LICENSE_TOKEN = ILicenseToken(licenseToken);
+    }
+
+    /// @notice Set the total license token limit for a specific license
+    /// @param licensorIpId The licensor IP id
+    /// @param licenseTemplate The license template address
+    /// @param licenseTermsId The license terms id
+    /// @param limit The total license token limit, 0 means no limit
+    function setTotalLicenseTokenLimit(
+        address licensorIpId,
+        address licenseTemplate,
+        uint256 licenseTermsId,
+        uint256 limit
+    ) external verifyPermission(licensorIpId) {
+        bytes32 key = keccak256(abi.encodePacked(licensorIpId, licenseTemplate, licenseTermsId));
```

Comment:
Do we want to add a check here to revert if the new limit is greater than current total supply?